### DISCUSSION TITLE
Indexer and dashboard password change causes logs/alert loss

### DISCUSSION THREAD

**Discussion by bleszily** [wazuh/wazuh#14530] [Q&A] [answered]
Hi, I have Wazuh deployed in AWS EKS with:
wazuh-dashboard-849dc4b767-kzx98
wazuh-indexer 3 replicas
wazuh-logstash
wazuh-manager-master 1 replica
wazuh-manager-worker 2 replicas
Whenever I change the passwords of the indexer and dashboard in the config files and apply changes, I have issues with losing all logs and alerts even when I have persistent volume and persistent volume claims configured. Please, any Idea what I am doing wrong?

Thanks.

**Comment by mdiego92** [comment, marked as answer]
Hello bleszily,

- Could you please confirm if your alerts are deleted from your Wazuh-Manager nodes after changing passwords?

`du -h /var/ossec/logs/alerts/2022/
`

- What's the current process you're following to set the new passwords?

Please remember that if you change the passwords of the internal users who work between the different services, you could have a negative impact on the proper functioning of the applications.

Sorry for the late response and thank you very much.

**Reply by bleszily** [reply]
Hi mdiego92,

Thanks for the reply. I only have alerts since the last password change. I lost all other alerts. This is the process I followed:
- Change the password on the secrets file (use the base64 command)
         * Dashboard and Indexer
- Create a new hash of the password using the script found on the indexer node (find / -name hash.sh)
	* export JAVA_HOME=/usr/share/wazuh-indexer/jdk	
	* cd /usr/share/wazuh-indexer/plugins/opensearch-security/tools
	* ./hash.sh -p <password>
- Change the newly created hash on the internal_users in the indexer file
	* Path to internal: wazuh/k8s/wazuh/indexer_stack/wazuh-indexer/indexer_conf/internal_users.yml
- Create new certificates using scripts: { Under certs}
	* Dashboard (remove all certs and regenerate)
	* Indexer (remove all certs and regenerate)
- Apply changes to Kubernetes
- Verify alerts:
	* du -h /var/ossec/logs/alerts/2022/
I see I am not supposed to change the internal user password. not sure if that's why I had this issue.

**Reply by mdiego92** [reply]
Hello bleszily,

Did you run the securityadmin.sh script for the certs? You will need to set up the paths if you have changed them.

JAVA_HOME=/usr/share/wazuh-indexer/jdk/

bash /usr/share/wazuh-indexer/plugins/opensearch-security/tools/securityadmin.sh -cd /usr/share/wazuh-indexer/plugins/opensearch-security/securityconfig/ -nhnv -cacert /usr/share/wazuh-indexer/config/certs/root-ca.pem -cert /usr/share/wazuh-indexer/config/certs/admin.pem -key /usr/share/wazuh-indexer/config/certs/admin-key.pem -p 9300 -icl

I'm really sorry for the late response again. I tried to reproduce your environment to perform these tests.

Thank you very much.

**Comment by mdiego92** [comment]
Hello bleszily,

Per our documentation, we recommend performing the password change for the users with the following procedure:
https://documentation.wazuh.com/current/user-manual/securing-wazuh/wazuh-indexer.html

There is no need to regenerate certificates. Neither to edit configuration files unless you change the password of an internal user who works between the different services as I said previously.

Also, could you please try recovering your alert data with our `recovery.py` script?
https://wazuh.com/blog/recover-your-data-using-wazuh-alert-backups/

Please let us know if you have any remaining questions or issues.

Thank you very much!

**Reply by bleszily** [reply]
Hi mdiego92

MY deployment is done on Kubernetes. not sure it's okay to use the password change script since each change will destroy old pods and create new ones. I will review the documentation. I have 1 master, 2 workers, and 3 indexer pods.

Regards.
